src/main/java/io/cryostat/core/net/JFRConnection.java

```diff
@@ -146,6 +152,36 @@ public synchronized int getPort() {
         }
     }
 
+    public synchronized int getJvmId() {
+        try {
+            if (!isConnected()) {
+                connect();
+            }
```

Comment:
For the other helper functions like `getApproximateServerTime`, `getHost`, how is `rjmxConnection` initialized when Cryostat is connecting to target JVMs?

I found out `this.rjmxConnection` is sometimes null at this point when starting recordings with automated rules. If call `connect()` here, the jvmId returns successfully, but I think my connection is conflicting with `connection.getService().start(recordingOptions, enableEvents(...)` from `RecordingTargetHelper.java:526`, resulting in a `NoSuchElementException` whenever `enableEvents` is called:
```
SEVERE: Exception thrown
java.util.NoSuchElementException: No value present
	at java.base/java.util.Optional.get(Optional.java:148)
	at io.cryostat.recordings.RecordingTargetHelper.enableEvents(RecordingTargetHelper.java:526)
	at io.cryostat.recordings.RecordingTargetHelper.lambda$1(RecordingTargetHelper.java:168)
	at io.cryostat.net.TargetConnectionManager.executeConnectedTask(TargetConnectionManager.java:156)
	at io.cryostat.recordings.RecordingTargetHelper.startRecording(RecordingTargetHelper.java:146)
	at io.cryostat.rules.RuleProcessor.lambda$6(RuleProcessor.java:275)
	at io.cryostat.net.TargetConnectionManager.executeConnectedTask(TargetConnectionManager.java:156)
	at io.cryostat.rules.RuleProcessor.startRuleRecording(RuleProcessor.java:259)
	at io.cryostat.rules.RuleProcessor.activate(RuleProcessor.java:178)
	at io.cryostat.rules.RuleProcessor.lambda$2(RuleProcessor.java:137)
```

Reply:
Ah my bad, it turns out this was a side effect of https://github.com/cryostatio/cryostat-web/issues/433 - everything is working now.